### PR TITLE
[k8s] Fix show-gpus when limited permissions are available

### PR DESCRIPTION
**Pull request by romilbhardwaj**
Updates our minimal permissions doc to add permissions required by show-gpus, and updates show-gpus to work with limited permissions:

```
W 10-29 18:36:09 kubernetes_catalog.py:112] Failed to get pods in the Kubernetes cluster (forbidden). Please check if your account has necessary permissions to list pods. Realtime GPU availability information may be incorrect.
Kubernetes GPUs (context: gke_sky-dev-465_us-central1-c_gkeusc6)
GPU   REQUESTABLE_QTY_PER_NODE  TOTAL_GPUS  TOTAL_FREE_GPUS
H100  1, 2, 4, 8                8           NA

Kubernetes per node GPU availability
NODE_NAME                               GPU_NAME  TOTAL_GPUS  FREE_GPUS
gke-gkeusc6-default-pool-db727de5-62gm  H100      8           NA
```

Also adds a quick snippet for specifying custom resources in pod_config.

### CODE REVIEW

**Comment by concretevitamin** [comment]
UX: `NA` may suggest "not available". How about `<insufficient permissions>`?

**Comment by romilbhardwaj** [comment]
`<insufficient permissions>` might be too long... how about `<no permissions>`? Or even `<forbidden>`?

**Comment by concretevitamin** [comment]
<no permissions> sounds good.

**Comment by romilbhardwaj** [comment]
Updated:
```
(base) ➜  ~ sky show-gpus --cloud kubernetes
W 11-11 14:06:42 kubernetes_catalog.py:114] Failed to get pods in the Kubernetes cluster (forbidden). Please check if your account has necessary permissions to list pods. Realtime GPU availability information may be incorrect.
Kubernetes GPUs (context: kind-skypilot)
GPU   REQUESTABLE_QTY_PER_NODE  TOTAL_GPUS  TOTAL_FREE_GPUS
H100  1, 2, 4, 8                8           <no permissions>

Kubernetes per node GPU availability
NODE_NAME               GPU_NAME  TOTAL_GPUS  FREE_GPUS
skypilot-control-plane  H100      8           <no permissions>
```

File: docs/source/reference/kubernetes/kubernetes-getting-started.rst [hunk 1]
```diff
@@ -296,6 +296,11 @@ FAQs
                 volumeMounts:       # Custom volume mounts for the pod
                   - mountPath: /foo
                     name: example-volume
+                resources:          # Custom resource requests and limits
+                  requests:
+                    rdma/rdma_shared_device_a: 1
+                  limits:
+                    rdma/rdma_shared_device_a: 1
```
Comment (Michaelvll) [inline docs/source/reference/kubernetes/kubernetes-getting-started.rst:303]: Is this relates to the `show-gpus`?

**Review by Michaelvll — APPROVED** [review]
Thanks for the update @romilbhardwaj! LGTM.

File: docs/source/reference/kubernetes/kubernetes-getting-started.rst (hunk 1 above)
Comment (romilbhardwaj) [reply docs/source/reference/kubernetes/kubernetes-getting-started.rst:303]: No, just adding docs on how to add custom resources through pod_config. Could have been a separate PR :)

**Pull request merged**